### DISCUSSION TITLE
[Feature] Narrow faces shown in search filter by typing (part of) the name

### DISCUSSION THREAD

**Discussion by thariq-shanavas** [immich-app/immich#7481] [Feature Request]
### The feature

I have close to a hundred faces with names assigned. This makes it very tedious to narrow a search by people in the search filter, as I need to scroll through all the faces and click on the ones I want. I think it would be an improved user experience if we could narrow down the faces shown in the search filter by typing part of the name in a text box. Here is a mock UI showing the "person search" on the top right.

![image](https://github.com/immich-app/immich/assets/19273195/d9888b1f-85dc-4153-b3b8-3f035fe5c387)

This "Narrow faces by typing part of the name" feature is already implemented in the people tab. As you type a name in, fewer options are shown which makes it easier to find the person you're looking for.

![image](https://github.com/immich-app/immich/assets/19273195/cc14c090-2c6f-4b2f-b8fb-ff885718963f)

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by github-cli** [comment]
would be nice to be able to select more than one person and combine them with "and/or/not" (or rather have and/or/not in general and have people just be another search term)

**Reply by thariq-shanavas** [reply]
You can already select more than one person! By default, it uses an 'and' filter, i.e., all selected people must appear in the search result.
I think having general and/not/or filters is too programmatic and would be confusing to the average user.

**Reply by github-cli** [reply]
it doesnt confuse the people using google (though with additional natural language search even iirc)...
I mean those that would be confused by it, wouldnt have the idea of using it

**Comment by thariq-shanavas** [comment]
@alextran1502  Following up with our conversation from discord here.
If you're willing to consider a bigger re-design of the search by face filter, may I suggest not showing all the face thumbnails at once? See a rough sketch of the mock UI below.

![image](https://github.com/immich-app/immich/assets/19273195/e868a15c-e7b7-4958-90db-acb0ce05ed40)


As the user starts typing a name, Immich shows suggestions.

![image](https://github.com/immich-app/immich/assets/19273195/170f898f-f4ce-45c2-918e-7963b0fe6bc1)


Once the user clicks on a name, Immich shows the selected faces as thumbnails on the right of the filter. If possible, we can have a "close" button to remove the selected person from the search.

![image](https://github.com/immich-app/immich/assets/19273195/8e763107-555f-4070-a683-a6e18ca7cccd)


Let me know your thoughts!
